src/curl/http_client.cpp

```diff
@@ -309,4 +315,25 @@ namespace ppconsul { namespace curl {
             throwCurlError(err, m_errBuffer);
     }
 
+    void CurlHttpClient::setHeaders(const http::RequestHeaders & headers)
+    {
+        m_headers.reset(nullptr);
```

Comment:
Let’s make the headers list into a local variable and assign it to the member only at the very end, so we don’t leave a partially built thing left behind in case of an error.

Reply:
Well it's going to cleanup on the next request anyways, isn't it? And exception is thrown in case of an error.

Reply:
That’s true. But also there is no reason to keep this object alive till next request in case of an error.
Ideally this should be split into (free) function that creates curl linked list from headers map and assignment its result to m_headers. This decomposition allows to test and reuse the function if needed. This correctly split responsibilities as well.

Reply:
The function I'm talking about may have the following signature:

```cpp
CurlHeaderList makeHeaderList(const http::RequestHeaders& headers)
```

Reply:
Oh, i was gonna change it but you closed PR :)

Reply:
Oh, nevermind. I've added this comment before I've seen that you've sent a CR and it was just a clarification of what signature the function may have.

Reply:
Sorry for confusion